Garrett-Iannuzzi/palette-picker
# Testing initial

#### What does this PR do?
- This PR implements the BE testing suite.
 -- It installs as dependencies: jest, cors and supertest in order to run the tests.
 -- It creates a test environment in our knexfile.js to run when we are implementing our tests. This also included creating a second seed file in a test directory to run within this environment. 
 -- Separates our app and endpoints into an app.js file so that we can test the endpoints in isolation from the creation and running of the server. The creation of the port and running of the server remains in the server.js file, which requires app from app.js to connect the two.
 -- Implements happy and sad path testing for all of our endpoints.
 -- Creates two new migrations that have been implemented:
 1. Deletes the projects_id columns in our palettes database
2. Re-creates the projects_id column in our palettes database to still be a foreign key referencing the projects.id, but also adds the `.onDelete('CASCADE')` functionality to implement cascading delete on the palette. So, if a project gets deleted all palettes with that project_id associated with it will also be deleted.

#### How to test?
- Pull down `testing-initial` branch 
- run `npm install`
- run `npm test`
- see app.js file to see actual tests

#### Issues:
- No issues have been closed
- The project board has been updated 

#### Notes:
- No new Notes on this one.

Thanks!
@Garrett-Iannuzzi
